benchopt/utils/seed_mixin.py

```diff
@@ -0,0 +1,46 @@
+import hashlib
+
+
+def get_seed(seed_dict, use_objective, use_dataset,
+             use_solver, use_repetition):
+    use_keys = {
+        "base_seed": True,
+        "objective": use_objective,
+        "dataset": use_dataset,
+        "solver": use_solver,
+        "repetition": use_repetition
+    }
+    hash_list = []
+    for key in use_keys:
+        if key not in seed_dict:
+            raise ValueError(f"Seed dict is not initialized correctly, "
+                             f"missing {key} key.")
+        elif use_keys[key]:
+            hash_list.append(seed_dict[key])
+        else:
+            hash_list.append("*")
+
+    hash_string = "_".join(hash_list)
+    digest = hashlib.sha256(hash_string.encode()).hexdigest()
+    seed = int(digest, 16) % (2**32 - 1)
+    return seed
+
+
+class SeedMixin:
+    """Mixin class to manage random seed for reproducibility."""
+
```

Comment:
Actually, something easier than parsing with `ast` might be:
```suggestion

    _use_seed = False
```

and then if `get_seed` is called, set `_use_seed = True`?